codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FlowController.java

```diff
@@ -22,6 +22,12 @@
 public interface Http2FlowController {
 
     /**
+     * Gets the {@link Http2Connection.PropertyKey} used for accessing the {@link Http2Stream.FlowControlState} property
+     * that this controller associates with each {@link Http2Stream}.
+     */
+    Http2Connection.PropertyKey stateKey();
```

Comment:
So I'm not saying "static across all connection" is the best/only option but it does have the following benefits:

1. Users of these keys can have static state if it makes sense.  The current API prevents static stuff and property keys from mixing together.
2. We don't need this interface on `Http2FlowController` :).
3. Potentially less object allocations per connection if the use case supports it (like in the flow controller case....these will always exist so why create 2 new key objects every time).